src/routes/ConfigureGame/index.tsx

```diff
@@ -1,165 +1,313 @@
-import React, { useState } from 'react';
-import classNames from 'classnames';
-import { Button, Checkbox, Icon, Input } from 'semantic-ui-react';
+import React, { useState, useEffect } from 'react';
+import { Title, Divider, Button, ActionIcon, Text, Checkbox, Grid, Group, Container } from '@mantine/core';
 import { useStore } from '../../useStore';
-import styles from './styles.module.css';
 import { useForm, FieldValues } from 'react-hook-form';
 import { FormInput } from '../../components/FormInputs';
 import { useParams, useNavigate } from 'react-router';
 import { Team } from '../../types';
 import { getEmptyTeam } from '../../helpers';
 import TeamGenerator from '../../components/TeamGenerator';
 import { Helmet } from 'react-helmet';
+import Icon from '../../components/Icon';
+import styles from './styles.module.css';
+import classNames from 'classnames';
 
-const formDefaultValues: { teams: Team[] } = {
+interface DefaultValue {
+  teams: Team[];
+  timeLimit: null | number;
+  selectionTimeLimit: null | number;
+  isQuestionPointsHidden: boolean;
+  mode: string;
+  players: string[];
+}
+
+const formDefaultValues: DefaultValue = {
   teams: [0, 1].map(() => getEmptyTeam()),
+  timeLimit: null,
+  selectionTimeLimit: null,
+  isQuestionPointsHidden: false,
+  mode: 'manual',
+  players: [],
 };
 
-export default function ConfigureGame() {
-  const { quizId, userName } = useParams();
+export default function ConfigureGame({ quizId }) {
+  const { userName } = useParams();
   const navigate = useNavigate();
   const {
     register,
     handleSubmit,
     formState: { errors },
-    getValues,
     setValue,
+    getValues,
+    watch,
```

Comment:
This is nice. Should we use it everywhere else, instead of getValues?

Reply:
this is readOnly, so using watch only the place where we are not updating the value